synapse/tests/test_lib_stormtypes.py

```diff
@@ -917,7 +919,7 @@ async def test_storm_path(self):
             self.isin('No var with name: testkey', err[1].get('mesg'))
 
             opts = {'vars': {'testkey': 'testvar'}}
-            text = "[ test:str='123' ] $path.vars.$testkey = test [ test:str=$testvar ]"
+            text = "[ test:str='123' ] $path.vars.$testkey = test [ test:str=$path.vars.testvar ]"
```

Comment:
Does this no longer work?

Reply:
yep, this is a rather extreme edge case, but essentially setting new (undeclared) vars via direct edits to `$path.vars.$foo = bar` makes them impossible to validate and would completely preclude doing undeclared variable detection.

Reply:
We can still support setting (and getting ) path specific vars this way, but can't handle hard-coded use of the var that was set dynamically ( why would you ever do that anyway :D )

Reply:
This is fine - just wanted to be explicit.  The directly manipulation of path and runtime vars has definitely always been a "here be dragons" territory.